### PR TITLE
repokitteh: add support for randomized auto-assign.

### LINKED ISSUE #13350: Assign API shepherds automatically via RepoKitteh
Load balance @envoyproxy/api-shepherds via RepoKitteh. This will work as follows:
1. An `auto_assign` attribute will be added to the existing repokitteh.star specs.
2. When a PR is updated, the set of GH users in owners will be computed, call this Owners.
3. We will compute the union of the assigness for the PR and requested reviewers for the PR, call this Assigned. 
4. If the intersection of Owners and Assigned is non-empty, there is no need for action.
5. Otherwise, a random member of Owners will be selected and a review requested. They will be nominated in the issue update comment as acting for the specced team.
This work is currently blocked on https://github.com/repokitteh/modules/issues/23. CC @itayd

### PR DESCRIPTION
**Pull request by htuch**
This will allow us to pick dedicated API shepherds for each PR, ensuring
we have clear review ownership.

Fixes #13350

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

**Comment by htuch** [comment]
This is now works and is ready for review, thanks.

**Comment by lizan** [comment]
side q: How this is automatically assigned without blessing rk? :D

**Comment by htuch** [comment]
@lizan yeah, I noticed that too. @itayd thoughts?

@lizan can I get a PR review as maintainer when you have a chance?

File: ci/repokitteh/modules/ownerscheck.star
Comment (lizan) [inline ci/repokitteh/modules/ownerscheck.star:208]: We should exclude the author from being auto assigned themself?

File: ci/repokitteh/modules/ownerscheck.star
Comment (htuch) [inline ci/repokitteh/modules/ownerscheck.star:162]: @itayd is this the right way to get at PR author? If not can you suggest an alternative method? Would be nice to have this keyword-literal args documented to be able to figure out semantics.

**Review by lizan — APPROVED** [review]
LGTM, finally seeing rk assigned someone other than me :)

File: ci/repokitteh/modules/ownerscheck.star
Comment (itayd) [reply ci/repokitteh/modules/ownerscheck.star:162]: i think it is the author. to be sure though you can access the raw event using the `event` parameter: `event.comment.user` (see for example https://prod.repokitteh.app/traces/ui/envoyproxy/envoy/814a4880-32c4-11eb-8650-62ce98aed958).

**Comment by itayd** [comment]
lgtm

**Pull request merged**
